src/traces/indicator/attributes.js

```diff
@@ -0,0 +1,408 @@
+var overrideAll = require('../../plot_api/edit_types').overrideAll;
+var fontAttrs = require('../../plots/font_attributes');
+var colorAttrs = require('../../components/color/attributes');
+var domainAttrs = require('../../plots/domain').attributes;
+var axesAttrs = require('../../plots/cartesian/layout_attributes');
+var templatedArray = require('../../plot_api/plot_template').templatedArray;
+var delta = require('../../constants/delta.js');
+
+var textFontAttrs = fontAttrs({
+    editType: 'plot',
+    colorEditType: 'plot'
+});
+
+var gaugeBarAttrs = {
+    color: {
+        valType: 'color',
+        editType: 'plot',
+        role: 'info',
+        description: [
+            'Sets the background color of the arc.'
+        ].join(' ')
+    },
+    line: {
+        color: {
+            valType: 'color',
+            role: 'info',
+            dflt: colorAttrs.defaultLine,
+            editType: 'plot',
+            description: [
+                'Sets the color of the line enclosing each sector.'
+            ].join(' ')
+        },
+        width: {
+            valType: 'number',
+            role: 'info',
+            min: 0,
+            dflt: 0,
+            editType: 'plot',
+            description: [
+                'Sets the width (in px) of the line enclosing each sector.'
+            ].join(' ')
+        },
+        editType: 'calc'
+    },
+    thickness: {
+        valType: 'number',
+        role: 'info',
+        min: 0,
+        max: 1,
+        dflt: 1,
+        editType: 'plot',
+        description: [
+            'Sets the thickness of the bar as a fraction of the total thickness of the gauge.'
+        ].join(' ')
+    },
+    editType: 'calc'
+};
+
+var stepsAttrs = templatedArray('threshold', extendDeep({}, gaugeBarAttrs, {
+    range: {
+        valType: 'info_array',
+        role: 'info',
+        items: [
+            {valType: 'number', editType: 'plot'},
+            {valType: 'number', editType: 'plot'}
+        ],
+        editType: 'plot',
+        // impliedEdits: {'autorange': false},
+        description: [
+            'Sets the range of this axis.',
+            'If the axis `type` is *log*, then you must take the log of your',
+            'desired range (e.g. to set the range from 1 to 100,',
+            'set the range from 0 to 2).',
+            'If the axis `type` is *date*, it should be date strings,',
+            'like date data, though Date objects and unix milliseconds',
+            'will be accepted and converted to strings.',
+            'If the axis `type` is *category*, it should be numbers,',
+            'using the scale where each category is assigned a serial',
+            'number from zero in the order it appears.'
+        ].join(' ')
+    }
+}));
+
+module.exports = {
+    mode: {
+        valType: 'flaglist',
+        editType: 'calc',
+        role: 'info',
+        flags: ['number', 'delta', 'gauge'],
+        dflt: 'number'
```

Comment:
`description` please.

Reply:
Done in e6d8525ab07c56ca10bdbac6762b06d1d57bf830